src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/hudsontrigger/actions/RetriggerAllAction.java

```diff
@@ -68,37 +68,24 @@ public RetriggerAllAction() {
 
     @Override
     public String getIconFileName() {
-        if (!hasPermission() || isBuilding() || !hasOthers()) {
-            return null;
-        } else {
-            return getPluginImageUrl("icon_retrigger24.png");
-        }
+        return getPluginImageUrl("icon_retrigger24.png");
     }
 
     @Override
     public String getDisplayName() {
-        if (!hasPermission() || isBuilding() || !hasOthers()) {
-            return null;
-        } else {
-            return Messages.RetriggerAll();
-        }
+        return Messages.RetriggerAll();
     }
 
     @Override
     public String getUrlName() {
-        if (!hasPermission() || isBuilding() || !hasOthers()) {
-            return null;
-        } else {
-            return "gerrit-trigger-retrigger-all";
-        }
+        return "gerrit-trigger-retrigger-all";
     }
 
     /**
      * Checks {@link TriggerContext#hasOthers()} if the context exists.
      * It does a null check on the context before calling.
      * @return true if there are any other builds in the context.
      */
-    @Restricted(NoExternalUse.class)
     public boolean hasOthers() {
```

Comment:
Why give permission for other plugins to call this method? The restriction is there so that the method can be removed if it is no longer needed. Was that your intention instead? Or perhaps you intended to make it `private`? All that is possible since it has been restricted for use in other plugins since it's inception :)

Reply:
ok, if this is the use of `@Restricted(NoExternalUse.class)` then yes, just marked the functions now as `private`.